### ISSUE TITLE
[BUG] openapi-generator 7.2.0 is no longer able to resolve local $refs

### ISSUE THREAD

**Issue by ctreatma** [OpenAPITools/openapi-generator#17554]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

We are attempting to upgrade our SDK generation process from openapi-generator 7.1.0 to 7.2.0, but 7.2.0 fails to find most of our OpenAPI spec; the output is littered with errors like this:

```
-Unable to load RELATIVE ref: ../../../components/schemas/CapacityList.yaml
```

The only change we're making is the openapi-generator version, so the files still exist in the same location; 7.1.0 is able to find them but 7.2.0 isn't.  After some poking it appears the issue is caused by the upgrade from swagger-parser 2.1.18 to 2.1.19.

##### Generation Details

I haven't had a chance to create a smaller reproduction case, but our upgrade attempt is here: https://github.com/equinix/equinix-sdk-go/pull/15

##### Steps to reproduce

Check out the PR linked above and run `make -f Makefile.metalv1 generate`.

##### Suggest a fix

I haven't identified the specific change in swagger-parser 2.1.19 that caused this bug, so there is currently no way to fix the bug by upgrading.  In the meantime, downgrading swagger-parser to 2.1.18 would fix this bug.

**Comment by vbauer** [comment]
It look like we have the same problem with version 7.2.0. Schemes located in the same file thats describing the operation are no longer visible to the parser:
```
[main] WARN  o.o.codegen.utils.ModelUtils - #/GetChatsResponse is not defined
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: #/GetChatsResponse
[main] WARN  o.o.codegen.utils.ModelUtils - #/GetChatsResponse is not defined
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: #/GetChatsResponse
[main] WARN  o.o.codegen.DefaultCodegen - Error obtaining the datatype from ref: #/GetChatsResponse. Default to 'object'
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: #/GetChatsResponse
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: #/GetChatsResponse
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: #/GetChatsResponse
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: #/GetChatsResponse
[main] WARN  o.o.codegen.utils.ModelUtils - #/GetChatsResponse is not defined
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: #/GetChatsResponse
```
Our specification: https://github.com/yandex-market/yandex-market-partner-api

Everything was fine in version 7.1.0

**Comment by ttrubinov** [comment]
I've created an issue in swagger-parser repo about this bug
https://github.com/swagger-api/swagger-parser/issues/2037

**Comment by ctreatma** [comment]
This was fixed in 7.3.0 by reverting the swagger-parser upgrade.  A Go test was added to attempt to catch similar issues caused by parser upgrades in the future.
